pkg/objstore/inmem/inmem.go

```diff
@@ -8,19 +8,23 @@ import (
 	"io/ioutil"
 	"strings"
 
+	"path"
+
 	"github.com/pkg/errors"
 )
 
 // Bucket implements the store.Bucket and shipper.Bucket interfaces against local memory.
 type Bucket struct {
 	objects map[string][]byte
+	dirs    map[string]struct{}
```

Comment:
Sorry, bit unrelated changes, but this actually makes inmem.Bucket behaviour the same as GCS one.